native/packages/react-native-bpk-component-badge/src/BpkBadge.js

```diff
@@ -157,6 +164,9 @@ const BpkBadge = (props: Props) => {
 
   return (
     <View style={viewStyle}>
+      {accessoryView && (
+        <View style={styles.accessoryView}>{accessoryView}</View>
+      )}
```

Comment:
This wrapper view is only being used to add the margin style. Could we avoid an extra view by conditionally applying the margin to the text instead, when the accessory view is supplied?

Reply:
I spoke to @willhg and we agreed that users shouldn't have to (or indeed be able to) colour the accessory view themselves, so the badge component should do this for them. 

We can do this by cloning the children and colouring any `Text` or `BpkIcon` components to match the badge's text colour.

Here's some examples of where we've iterated over or cloned children before that you could base it off:

1. [Cloning images passed to flat list items to add a margin](https://github.com/Skyscanner/backpack/blob/master/native/packages/react-native-bpk-component-flat-list/src/BpkFlatListItem.android.js#L74)
1. [Iterating over children passed into pickers to grab some data](https://github.com/Skyscanner/backpack/blob/master/native/packages/react-native-bpk-component-picker/src/BpkPickerMenu.android.js#L76)